qml/DepthAxis.qml

```diff
@@ -0,0 +1,128 @@
+import QtQuick.Layouts 1.3
+
+import Logger 1.0
+
+Item {
+    id: root
+
+    // these are actually all in meters right now
+    property var start_mm: 0
+    property var end_mm: 0
+    property var length_mm: end_mm - start_mm
+    property var depth_mm: 0
+
+    property var validIncrements: [0.1, 0.25, 0.5, 1, 2, 5, 10]
+
+    property int numTicks: 10
+    property int maxTicks: 10
+    property var increment: 1
+
+    property var color: "white"
+
+    function getIncrement() {
+        var inc;
+        for (var i = 0; i < validIncrements.length; i++) {
+            inc = validIncrements[i];
+            if (length_mm / inc <= maxTicks) {
+                break;
+            }
+        }
+        return inc;
+    }
+
+    function recalc() {
+        increment = getIncrement()
+        numTicks = length_mm / increment + 1
+    }
+
+    onStart_mmChanged: {
+        recalc()
+    }
+
+    onEnd_mmChanged: {
+        recalc()
+    }
+
+    function nominalHeight() {
+        return increment * parent.height / length_mm
+    }
+
+    function remainingHeight() {
+        return parent.height - ((numTicks-1) * nominalHeight())
+    }
+
+    Component {
+        id: tickMark
+        RowLayout {
+            anchors.right: parent.right
+            height: lastIndex ? remainingHeight() : nominalHeight()
+            Layout.fillHeight: true
+            Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+
+            property var lastIndex: (index + 1) == numTicks
+
+            Rectangle {
+                id: tick
+                width: 15
+                height: 2
+                Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+                color: root.color
+                border.color: root.color
+                border.width: 1
+                visible: parent.height > 0
+            }
+        }
+    }
+
+    Component {
+        id: tickLabel
+        RowLayout {
+            anchors.right: parent.right
+            height: lastIndex ? remainingHeight() : nominalHeight()
+            Layout.fillHeight: true
+            Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+
+            property var lastIndex: (index + 1) == numTicks
+
+            Text {
+                id: label
+                //anchors.fill:parent // this text object is not anchored
+                //Layout.fillHeight: true
+                Layout.alignment: Qt.AlignRight | Qt.AlignBottom
+                Layout.bottomMargin: 5
+                Layout.rightMargin: 7
+                text: lastIndex ? end_mm.toFixed(1) : start_mm + (index + 1) * increment
+                color: root.color
+                font.bold: true
+                font.pointSize: 22
+                visible: parent.height >= height
+            }
+        }
+    }
+
+    Image {
+        anchors.right: parent.right
+        source: "/icons/depth_indicator.svg"
+        height: 20
+        width: 20
+        y: (parent.height / length_mm) * depth_mm - height / 2
+    }
```

Comment:
For simple things like triangle, or square, you can use canvas:
```cpp
    Canvas {
        id: depthPointer
        y: (parent.height / length_mm) * depth_mm - height / 2
        z: root.z + 1
        width: 20
        height: width
        onWidthChanged: requestPaint()
        onPaint: {
            print(width, height)
            var ctx = getContext("2d")
            ctx.fillStyle = "red"
            ctx.beginPath()
            ctx.moveTo(width, 0)
            ctx.lineTo(0, height/2)
            ctx.lineTo(width, height)
            ctx.fill()
        }
    }
```

Reply:
That would require thought and effort

Reply:
This code already do the red triangle. You just need to copy and paste :)

Reply:
thanks for doing the hard work for me